src/basic/path-lookup.c

```diff
@@ -91,6 +91,41 @@ int xdg_user_data_dir(char **ret, const char *suffix) {
         return 1;
 }
 
+int runtime_directory(char **ret, const char *suffix) {
+        int r;
+        bool apply_suffix = true;
+        _cleanup_free_ char *d = NULL;
+
+        assert(ret);
+        assert(suffix);
+
+        /* Accept $RUNTIME_DIRECTORY as authoritative
+         * If its missing apply the suffix to /run or $XDG_RUNTIME_DIR
+         * depending on whether are root or a regular user.
+         *
+         * Return value indicates whether the suffix was applied or not */
+
+        const char *e = getenv("RUNTIME_DIRECTORY");
+        if (e) {
+                d = strdup(e);
```

Comment:
frankly, i'd just exit here.

i.e. if this works, exit early with success, returning what you just found. there's no reason to go through the rest of the function, as you have nothing to append anyway. i think the code would be easier and shorter that way.

Reply:
i think we better use secure_getenv() here, for paranoia. given that this is a really generic helper call